### ISSUE TITLE
`builtins-ignorelist` for attributes only

### ISSUE THREAD

**Issue by benjamin-kirkbride** [astral-sh/ruff#6524]
Looking at https://beta.ruff.rs/docs/rules/builtin-attribute-shadowing/#builtin-attribute-shadowing-a003: I agree with the point made here: https://stackoverflow.com/a/60423602/1342874

Specifically, I think that the following is fine, and in fact better than alternatives:
```python
@dataclass
class Row:
    id: int
    input: str
```

but of course the following is not fine: 
```python
id, input = row.id, row.input
```

My understanding is that the `builtins-ignorelist` is all or nothing; it makes it so that you can us `id` everywhere without warning, or nowhere without warning.

My proposal is that there is a `builtins-attributes-ignorelist` or something similar added for this.

**Comment by charliermarsh** [comment]
Why not disable the rule entirely? It's intentionally categorized as a separate rule to enable that kind of granular on/off switch.

**Comment by benjamin-kirkbride** [comment]
Fair question :sweat_smile: 

I think there are some attributes that I would still want to list, like `str`, or `dict` or something, but I don't have a good argument for why I think that some things are okay (like `id`, `input`, `class`, etc) but others aren't..

**Comment by tylerlaprade** [comment]
I ran across this as well. I really don't want `type` to be used as a Django model attribute because if I accidentally use `type` instead of `self.type`, my code will be completely wrong, but I won't get any warnings. But `id` and `filter` seem generally harmless for my usecases.

**Comment by Olegt0rr** [comment]
Just get `A003 Class attribute `exec` is shadowing a python builtin` on a method.

```python
class Foo:
    async def exec(self):
        ...
```

It's not a shadowing, cause I still can use builtin `exec`.

```python
class Foo:
    @staticmethod
    def exec():
        exec("print('it is not a shadowing')")

Foo.exec()

```

**Comment by charliermarsh** [comment]
@Olegt0rr - That _is_ working as intended but it's a strange rule. You can see an example of what it's intended to catch [here](https://github.com/astral-sh/ruff/issues/4470):

```python
class C:
    @staticmethod
    def list() -> None:
        pass

    @staticmethod
    def repeat(value: int, times: int) -> list[int]:
        return [value] * times
```

The issue is that it _will_ be shadowed if you reference the builtin within the class scope.

**Comment by charliermarsh** [comment]
PR up that should make this rule _far_ more useful by only flagging actual shadowed reads (like in the example above) rather than the method definitions (which will almost always be fine): https://github.com/astral-sh/ruff/pull/9462
